apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/RdaLoadOptions.java

```diff
@@ -54,17 +52,18 @@ public RdaLoadOptions(
    * @param appState the shared {@link PipelineApplicationState}
    * @return a PipelineJob instance suitable for use by PipelineManager.
    */
-  public PipelineJob<NullPipelineJobArguments> createFissClaimsLoadJob(
-      PipelineApplicationState appState) {
+  public RdaFissClaimLoadJob createFissClaimsLoadJob(
+      PipelineApplicationState appState, Clock clock) {
     return new RdaFissClaimLoadJob(
         jobConfig,
         () ->
             new GrpcRdaSource<>(
                 grpcConfig,
                 new FissClaimStreamCaller(
                     new FissClaimTransformer(Clock.systemUTC(), new IdHasher(idHasherConfig))),
```

Comment:
Is it intentional that the clock passed in as a parameter is not being used here as well? Passing all of these clock objects around as parameters seems like it could get out of hand and lead to strange behavior that is hard to track down. Don't we always want just one clock across every piece of code (at least for business logic)?

Reply:
The clock needs to be configurable for unit testing.  Ideally we would have a dependency manager that could provide dependencies like this but we don't.   I considered adding a clock to the `PipelineApplicationState` but thought others might object.  If you'd find that preferable I could make that change.  We really should not use a singleton for the clock because those are much more error prone than just passing a dependency to a constructor.

Reply:
I was not thinking singleton either but more along the lines of dependency injection but if we don't have that available I think the PipelineApplicationState would be a reasonable compromise.

Can you enlighten me as to whether this particular line is using the wrong clock? Or is there some reason why we want to use the system clock here even though we were passed a clock as a parameter that might be different?

Reply:
LOL - yeah that's a bug.

Reply:
Got it. Same bug also exists a few lines down.

Reply:
Bug fixed and I've moved the Clock into PipelineApplicationState. That removed it from constructors that accept it.